eng/references.props

```diff
@@ -1,10 +0,0 @@
-<Project>
-  <PropertyGroup>
-    <!--
-      The Microsoft.NET.Sdk defaults the AssemblySearchPaths earlier, so default
-      the correct search paths here. We are rather restrictive by design - only ref assemblies should
-      be resolved by default.
-    -->
-    <AssemblySearchPaths>$(AssemblySearchPaths);$(RefPath);{RawFileName}</AssemblySearchPaths>
```

Comment:
Why are we removing this?  IIRC this was done to avoid some undesireable probing locations like the GAC.

Reply:
We already have this https://github.com/dotnet/runtime/blob/master/eng/references.targets#L5